src/libraries/Microsoft.PowerFx.Core/Public/Values/DelegationParameters.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Functions;
+
+namespace Microsoft.PowerFx.Types
+{
+    /// <summary>
+    /// For use with <see cref="TableValue"/> and delegation.
+    /// Describe delegation parameters such as filters, column selection, and sorting. 
+    /// </summary>
+    public abstract class DelegationParameters
```

Comment:
Delegation analysis logic will convert queries `Filter(Accounts, Age > 30)` into DelegationParameter  and then invoke on IDelegatableTableValue.

Reply:
Also tbd - we could consider defining in Interpreter (isntead of Core), since that's reasonably common.